practice_points.c

```diff
@@ -0,0 +1,107 @@
+#include <cstring>
+#include <memory>
```

Comment:
C-compiler friendly.
```suggestion
#include <stdio.h>
#include <stdlib.h>

#pragma warning ( disable : 4996 ) // strok has been replaced by strok_s in C11, VS2019 considers it unsafe
```